protocol/messenger_store_node_request_manager.go

```diff
@@ -0,0 +1,327 @@
+package protocol
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/communities"
+	"github.com/status-im/status-go/protocol/transport"
+)
+
+const (
+	storeNodeAvailableTimeout = 5 * time.Second
+)
+
+// FetchCommunityStats is used in tests
+type FetchCommunityStats struct {
+	FetchedEnvelopesCount int
+	FetchedPagesCount     int
+}
+
+type StoreNodeRequestManager struct {
+	messenger *Messenger
```

Comment:
Would it make sense to use interface instead to break bidirectional dependency?

Reply:
So I checked this. Here's a list of Messenger functions used here:
- `.transport.FilterByChatID`
- `.transport.InitPublicFilters`
- `transport.RemoveFilters`
- `.passStoredCommunityInfoToSignalHandler`
- `.ProcessAllMessages`
- `.communitiesManager.GetByIDString`
- `.waitForAvailableStoreNode`
- `.GetCurrentTimeInMillis`
- `.performMailserverRequest`
- `.processMailserverBatchWithOptions`

For me it looks like it's better to keep bidirectional dependency than having such an interface :smile:
Though I'm sure this can be simplified to less methods.

I'll leave it as is for now and see where what we have after implementing contact fetch in the same way. It will makes more sense to refactor after that.